constellations/orion/src/controllers/classes_by_semesters_controller.go

```diff
@@ -0,0 +1,118 @@
+package controllers
+
+import (
+	"net/http"
+
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/domains"
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/repos"
+	"github.com/gin-gonic/gin"
+)
+
+func GetAllProgramsSemestersClasses(c *gin.Context) {
+	var classSemesterJson domains.ProgramClassesBySemester
+	var listResults []domains.ProgramClassesBySemester
+
+	// Fetch progrmas, semesters, classes from repo functions
+	publishedOnly := ParseParamPublishedOnly(c)
+
+	programs, err := repos.ProgramRepo.SelectAll(publishedOnly)
+	semesters, err := repos.SemesterRepo.SelectAll(publishedOnly)
+	classes, err := repos.ClassRepo.SelectAll(publishedOnly)
+
+	// Convert lists into maps
+	programMap := make(map[string]domains.Program)
+	semesterMap := make(map[string]domains.Semester)
+	classMap := make(map[string]domains.Class)
+
+	for i := 0; i < len(programs); i++ {
+		programMap[programs[i].ProgramId] = programs[i]
+	}
+	for i := 0; i < len(semesters); i++ {
+		semesterMap[semesters[i].SemesterId] = semesters[i]
+	}
+	for i := 0; i < len(classes); i++ {
+		classMap[classes[i].ClassId] = classes[i]
+	}
+
+	// Semester to class list map (maps semesterId to Class object)
+	semesterClassMap := make(map[string][]domains.Class)
+
+	// loop over semesterIds
+	for _, value := range semesterMap {
+		// create Semester Object
+		semesterObj := value
+		var programClassStruct domains.ProgramClass
+		var programObj domains.Program
+
+		// Find classes that have the same semester Id and append to semesterClassMap
+		classSlice := make([]domains.Class, 0)
+		for i := 0; i < len(classes); i++ {
+			if classes[i].SemesterId == semesterObj.SemesterId {
+				classSlice = append(classSlice, classes[i])
+			}
+		}
+		semesterClassMap[semesterObj.SemesterId] = classSlice // list of classes in specific semester
+
+		// create list of programIds in specific semester
```

Comment:
Capitalize first word of all comments i.e.
Create list of programIds in specific semester